### PR TITLE
Flowize linter

### PR DESCRIPTION
**Pull request by steelbrain**
Flowize the whole package

### CODE REVIEW

File: lib/commands.js [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
+import type { Disposable } from 'atom'
 
 export default class Commands {
+  emitter: Emitter;
+  subscriptions: CompositeDisposable;
+
   constructor() {
-    this.subscriptions = new CompositeDisposable()
     this.emitter = new Emitter()
+    this.subscriptions = new CompositeDisposable()
```
Comment (Arcanemagus) [inline lib/commands.js]: Not that there is really a _problem_ with this, but why re-order these?

**Comment by Arcanemagus** [comment]
LGTM

File: lib/commands.js (hunk 1 above)
Comment (steelbrain) [inline lib/commands.js]: I am moving the shorter variables above the longer ones so they look nice

**Comment by joefitzgerald** [comment]
:-1: on this. What was the benefit? When was it decided to use flow on the package?

**Comment by steelbrain** [comment]
@joefitzgerald The main benefit is that flow is smart, so if we have a class like this
```js
class Registry {
  add(something: Object)
}
```
and try to do
```js
registry.aa(someObj);
```
or
```js
registry.add('something')
```
it will error and tell us that we're doing it wrong.

It has been extremely helpful and it's autocomplete has been very accurate. We catch bugs everytime we move a package to flow. Can you state the reasons you are against this change?

**Pull request merged**